### PR TITLE
fix: formatting code

### PR DESCRIPTION
**Pull request by Francesco-Ranieri**

### CODE REVIEW

**Comment by Francesco-Ranieri** [comment]
Proposed structure 

📦mm-cot
 ┣ 📂data
 ┃ ┣ 📂dataset
 ┃ ┃ ┣ 📂scienceqa
 ┃ ┃ ┃ ┣ 📜pid_splits.json
 ┃ ┃ ┃ ┗ 📜problems.json
 ┃ ┃ ┗ 📜captions.json
 ┃ ┗ 📂vision_features
 ┃ ┃ ┣ 📜clip.npy
 ┃ ┃ ┣ 📜detr.npy
 ┃ ┃ ┣ 📜mm-cot.png
 ┃ ┃ ┣ 📜name_map.json
 ┃ ┃ ┗ 📜resnet.npy
 ┣ 📂docs
 ┃ ┣ 📂images
 ┃ ┃ ┗ 📜mm-cot.png
 ┃ ┣ 📜CODE_OF_CONDUCT.md
 ┃ ┗ 📜CONTRIBUTING.md
 ┣ 📂models
 ┃ ┣ 📂MM-CoT-UnifiedQA-base-Answer
 ┃ ┃ ┣ 📜.DS_Store
 ┃ ┃ ┣ 📜config.json
 ┃ ┃ ┣ 📜predictions_ans_test.json
 ┃ ┃ ┣ 📜pytorch_model.bin
 ┃ ┃ ┣ 📜special_tokens_map.json
 ┃ ┃ ┣ 📜spiece.model
 ┃ ┃ ┣ 📜tokenizer_config.json
 ┃ ┃ ┗ 📜training_args.bin
 ┃ ┗ 📂MM-CoT-UnifiedQA-base-Rationale
 ┃ ┃ ┣ 📜.DS_Store
 ┃ ┃ ┣ 📜config.json
 ┃ ┃ ┣ 📜gate.txt
 ┃ ┃ ┣ 📜predictions_ans_eval.json
 ┃ ┃ ┣ 📜predictions_ans_test.json
 ┃ ┃ ┣ 📜pytorch_model.bin
 ┃ ┃ ┣ 📜special_tokens_map.json
 ┃ ┃ ┣ 📜spiece.model
 ┃ ┃ ┣ 📜tokenizer_config.json
 ┃ ┃ ┗ 📜training_args.bin
 ┣ 📂notebooks
 ┃ ┗ 📜Multimodal_Chain_of_Thought_Reasoning_in_Language_Models.ipynb
 ┣ 📂src
 ┃ ┣ 📂data
 ┃ ┃ ┣ 📜data.py
 ┃ ┃ ┗ 📜__init__,py
 ┃ ┣ 📂models
 ┃ ┃ ┣ 📜evaluation.py
 ┃ ┃ ┣ 📜evaluations.py
 ┃ ┃ ┣ 📜model.py
 ┃ ┃ ┣ 📜prompt.py
 ┃ ┃ ┣ 📜t5_trainer.py
 ┃ ┃ ┗ 📜__init__,py
 ┃ ┣ 📜main.py
 ┃ ┣ 📜parser_args.py
 ┃ ┗ 📜__init__,py
 ┣ 📜.gitignore
 ┣ 📜init_project.py
 ┣ 📜README.md
 ┣ 📜requirements.txt
 ┣ 📜run_inference.sh
 ┣ 📜run_training.sh
 ┗ 📜setup.py

File: run_inference.sh [hunk 1]
```diff
@@ -1,17 +1,18 @@
...
+  --bs 8 \
+  --eval_bs 4 \
+  --eval_acc 10 \
+  --output_len 512 \
+  --final_eval \
+  --prompt_format QCM-LE \
+  --evaluate_dir models/rationale \
+  --data_root data/dataset \
+  --caption_file data/dataset/captions.json \
+  --evaluate_dir models/MM-CoT-UnifiedQA-base-Rationa
```
Comment (gianfrancodemarco) [inline run_inference.sh]: why did you add these?

File: run_inference.sh [hunk 2]
```diff
@@ -1,17 +1,18 @@
...
 
 # answer inference
-CUDA_VISIBLE_DEVICES=0,1 python main.py \
-    --model allenai/unifiedqa-t5-base \
-    --user_msg answer --img_type detr \
-    --bs 8 --eval_bs 4 --eval_acc 10 --output_len 64 \
-    --final_eval --prompt_format QCMG-A \
-    --eval_le models/rationale/predictions_ans_eval.json \
-    --test_le models/rationale/predictions_ans_test.json \
-    --evaluate_dir models/answer
```
Comment (gianfrancodemarco) [inline run_inference.sh]: why did you remove this?

File: run_training.sh [hunk 3]
```diff
@@ -1,15 +1,6 @@
...
+
 
 # answer inference
-CUDA_VISIBLE_DEVICES=0,1 python main.py \
-    --model allenai/unifiedqa-t5-base \
-    --user_msg answer --img_type detr \
-    --bs 8 --eval_bs 4 --eval_acc 10 --output_len 64 \
-    --final_eval --prompt_format QCMG-A \
-    --eval_le experiments/rationale_allenai-unifiedqa-t5-base_detr_QCM-LE_lr5e-05_bs16_op512_ep20/predictions_ans_eval.json \
-    --test_le experiments/rationale_allenai-unifiedqa-t5-base_detr_QCM-LE_lr5e-05_bs16_op512_ep20/predictions_ans_test.json
```
Comment (gianfrancodemarco) [inline run_training.sh]: Why did you remove this?

File: src/main.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
+import json
+import os
+import random
+
+from rich import box
+from rich.table import Column, Table
+
+from src.data.data import load_data_std, load_data_img
+from src.models.t5_trainer import T5Trainer
+from src.parser_args import parse_args
```
Comment (gianfrancodemarco) [inline src/main.py]: Rename the file args_parser.py or parse_args.py

File: run_inference.sh (hunk 1 above)
Comment (Francesco-Ranieri) [reply run_inference.sh]: This changes allow the script execution because the default structure was changed.
We can change the default value of the arguments as follow:
data_root: from 'data' to 'data/dataset'
caption_file: from 'data/captions.json' to 'data/dataset/captions.json'

The last one, the evaluate_dir is necessary but it should be fixed because the rigth args is
--evaluate_dir models/MM-CoT-UnifiedQA-base-Rationa**le**

File: src/main.py (hunk 4 above)
Comment (Francesco-Ranieri) [reply src/main.py]: ok

File: run_inference.sh (hunk 1 above)
Comment (Francesco-Ranieri) [reply run_inference.sh]: Do you think it is better to update the args default value or to leave this script parameters?
If we change args default value there is no need to modify the other sh scripts

Comment (gianfrancodemarco) [reply run_inference.sh]: Ok, fix the last parameter and you can keep the others

**Review by gianfrancodemarco — APPROVED** [review]

**Comment by Francesco-Ranieri** [comment]
@gianfrancodemarco just keep in mind that, when we are able to run traning and interface scripts, we probabily need to add the args values like in the run_interface.sh
